Documentation/api.md

```diff
@@ -188,6 +192,62 @@ BasicAuth allow an endpoint to authenticate over basic authentication More info:
+| path | HTTP path to scrape for metrics. | string | false |
+
+[Back to TOC](#table-of-contents)
+
+## BlackboxMonitor
+
+BlackboxMonitor defines monitoring for a set of static targets or ingresses.
+
+| Field | Description | Scheme | Required |
+| ----- | ----------- | ------ | -------- |
+| metadata | Standard object’s metadata. More info: https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#metadata | [metav1.ObjectMeta](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#objectmeta-v1-meta) | false |
+| spec | Specification of desired Ingress selection for target discovery by Prometheus. | [BlackboxMonitorSpec](#blackboxmonitorspec) | true |
+
+[Back to TOC](#table-of-contents)
+
+## BlackboxMonitorList
+
+BlackboxMonitorList is a list of BlackboxMonitors.
+
+| Field | Description | Scheme | Required |
+| ----- | ----------- | ------ | -------- |
+| metadata | Standard list metadata More info: https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#metadata | [metav1.ListMeta](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#listmeta-v1-meta) | false |
+| items | List of BlackboxMonitors | []*[BlackboxMonitor](#blackboxmonitor) | true |
+
+[Back to TOC](#table-of-contents)
+
+## BlackboxMonitorSpec
+
+BlackboxMonitorSpec contains specification parameters for a BlackboxMonitor.
+
+| Field | Description | Scheme | Required |
+| ----- | ----------- | ------ | -------- |
+| jobLabel | The label to use to retrieve the job name from. | string | false |
+| blackboxExporterSpec | Specification for the blackbox exporter to use for probing targets. | [BlackboxExporterSpec](#blackboxexporterspec) | false |
+| module | The module to use for probing. | string | false |
+| staticTargets | List of static targets for the blackbox exporter. | []string | false |
+| selector | Select Ingress objects by labels. | [metav1.LabelSelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#labelselector-v1-meta) | false |
+| namespaceSelector | Select Ingress objects by namespace. | [NamespaceSelector](#namespaceselector) | false |
+| params | Optional HTTP URL parameters | map[string][]string | false |
+| interval | Interval at which metrics should be scraped | string | false |
+| scrapeTimeout | Timeout after which the scrape is ended | string | false |
+| honorLabels | HonorLabels chooses the metric's labels on collisions with target labels. | bool | false |
+| honorTimestamps | HonorTimestamps controls whether Prometheus respects the timestamps present in scraped data. | *bool | false |
+| metricRelabelings | MetricRelabelConfigs to apply to samples before ingestion. | []*[RelabelConfig](#relabelconfig) | false |
+| relabelings | RelabelConfigs to apply to samples before ingestion. More info: https://prometheus.io/docs/prometheus/latest/configuration/configuration/#relabel_config | []*[RelabelConfig](#relabelconfig) | false |
```

Comment:
Since the module is configurable I'm not sure metric relabeling or normal relabel configs are necessary here. Did you have a concrete use case in mind here?

Reply:
Actually yes. 
We have some scenarios in which an ingress may define multiple hosts, but we only want to probe hosts satisfying a certain regex; others hosts shall be ignored
For an Ingress like

```yaml
kind: Ingress
...

spec:
  rules:
    - host: probe.me
       ...
    - host: ignore.me
       ...
```

the following relabel config would be necessary to only probe certain hosts defined in the Ingress spec.

```yaml
- source_labels: [__address__]
  regex: .*probe.me.*
  action: keep
```

Reply:
Ah yes that makes sense. But metric relabeling wouldn't be needed right?

In either case I'm also thinking that static targets would need label definitions.

Reply:
Dropping metric relabeling.
Currently, one can define additional relabeling via `BlackoxMonitorSpec.RelabelingConfigs`, which are applied to both static and ingress sd targets.

Reply:
I'm saying static targets should be able to list their labels as well like [here](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#static_config). (not related to relabeling per se, just "labeling"). What do you think?

Reply:
I see. Then how about something like
```
spec: 
  # Relabelings for static & dyn. targets.
  relabelings:
    ...
  targets:
    static:
      targets: 
        - url.com
      # Labels for metrics from static targets.
      labels:
        key: value
        ...
    ingress:
      selector: {}
      namespaceSelector: {}
      # Labels for metrics from ingress' targets.
      labels:
        key: value
        ...
```

Reply:
I think relabel configs should really only be necessary for the dynamic parts (ingress/service discovery), so I think I would prefer to move it under that instead of directly under the spec. Also it was an oversight when we introduced it in the other objects to call it "relabelings", let's go with what it's called in prometheus and name it `relabelConfigs` (we'll want to rename these at some point in the other objects or in their next iteration).

Reply:
Sorry for the massive delay. Busy times.

It's now with `StaticConfig` for configuring static targets that shall be probed. Config matching the existing way of doing [this](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#static_config).
And changed `relabelings` to `RelabelConfigs` in the section for configuring dyn. discovered `Ingress` targets. 
Are you ok with that @brancz?

```
Spec: 
  Targets:
    StaticConfig:
      Targets: []string
      Labels: map[string]string
        ...
    Ingress:
      Selector: LabelSelector
      NamespaceSelector: NamespaceSelector
      RelabelConfigs: []*RelabelConfig
        ...
```

I need to fix the tests and rebase again but will push latest today or tomorrow.

Reply:
Exactly what I was thinking 🎉🎊